src/Events/Calendar_Embeds/Admin/Restore_Menu_Trait.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * Trait Restore_Menu_Trait
+ *
+ * @since TBD
+ *
+ * @package TEC\Events\Calendar_Embeds\Admin
+ */
+
+namespace TEC\Events\Calendar_Embeds\Admin;
+
+/**
+ * Trait Restore_Menu_Trait
+ *
+ * @since TBD
+ */
+trait Restore_Menu_Trait {
+
+	/**
+	 * The stored globals.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	protected static array $stored_globals = [
+		'parent_file'  => '',
+		'submenu_file' => '',
+	];
+
+	/**
+	 * Restores the current parent file.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function restore_menu_globals(): void {
+		if ( ! empty( self::$stored_globals ) ) {
```

Comment:
Should this one be 
```suggestion
		if ( empty( self::$stored_globals ) ) {
```

I don't see how we're going to get to the line 45 otherwise?

Reply:
Also - I'm not understanding why this is needed?